src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Loader } from '../Loader';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { OrderControl } from '../../utils/OrderControl';
+import { PeopleFilters } from '../PeopleFilters/PeopleFilters';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+
+const getPreparedPeople = (people: Person[]) => {
+  return people.map(person => {
+    const father = people.find(fath => fath.name === person.fatherName);
+    const mother = people.find(moth => moth.name === person.motherName);
+
+    return {
+      ...person,
+      father,
+      mother,
+    };
+  });
+};
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+
+  const [searchParams] = useSearchParams();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    getPeople()
+      .then((peopleFromServer) => {
+        setPeople(getPreparedPeople(peopleFromServer));
+      })
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const displayedPeople = () => {
+    let filteredPeople = [...people];
+
+    const centuries = searchParams.getAll('centuries') || [];
+    const gender = searchParams.get('sex') || '';
+    const query = searchParams.get('query') || '';
```

Comment:
suggest to make enum for all possible search params fields. and use in all places instead strings.